sopel/modules/admin.py

```diff
@@ -327,12 +327,22 @@ def set_config(bot, trigger):
         bot.say(exc.args[1])
         return
 
+    # Get a descriptor class for the option if it's a static section
+    descriptor = getattr(section.__class__, option) if static_sec else None
+
     # Display current value if no value is given
     if not value:
-        if option.endswith("password") or option.endswith("pass"):
+        value = getattr(section, option)
```

Comment:
Maybe it's a micro-optimization, but why assign twice if the option is a secret/password? What was wrong with the previous version's `else` below?

Reply:
Because it's not a simple `if/else` now, it's a `if (then if/else) / else` (nested if).

At first, `value` is initialized, and then the responsibility of the code that follows this initialization is to override it if necessary, and only to do that. It limits the responsibility of said code to a very small task, reducing its working complexity.

Basically, every if/else following this statement should do one and only on thing: check for & apply override. This is especially useful to reduce the number of `else` case to manage (and test).